### ISSUE TITLE
Cleaning up geometry module: moving Tipable & Arrows to a new module

### ISSUE THREAD

**Issue by AntonBallmaier** [ManimCommunity/manim#1537]
### New Module
The `geometry` module is extremely crowded (2558 lines, 39 classes). 

Therefor I suggest moving the classes connected to arrows to a now module, which could be named  `tipable.py`.
This should includes the following:

- TipableVMobject
- Arrow
- DoubleArrow
- CurvedArrow
- CurvedDoubleArrow
- ArrowTip 
- ArrowCircleFilledTip
- ArrowCircleTip
- ArrowSquareFilledTip
- ArrowSquareTip
- ArrowTriangleFilledTip
- ArrowTriangleTip

### Combining classes
The number of classes is quite a bit over the top in my opinion. I feel like adding an aditional parameter wold be a better choice in many cases:

- Giving Arrow two parameters `double` and `curved` both defaulting to `False` would deprecate  DoubleArrow, CurvedArrow and CurvedDoubleArrow (Imo whether CurvedArrow should be deprecated is debatable).
- The ArrowTip should get a parameter `filled` defaulting to `False`. This would deprecate the three filled variant classes.

### Class renamings:
The (remaining) arrow tip classes ought to be renamed:

- ArrowCircleTip  => CircleArrowTip
- ArrowSquareTip => SquareArrowTip
- ArrowTriangleTip => TriangleArrowTip, however it might also be a good idea to combine TriangleArrowTip and ArrowTip in ArrowTip. 

## Additional comments
Please add a comment if you want to work at this issue, as it may well be I already started doing so :)

**Comment by behackl** [comment]
CC @XorUnison, you mentioned at some point that you are looking to rework arrows?

**Comment by AntonBallmaier** [comment]
Ok, I hope it is ok for me to start working on this,  as his "GeometryAdditions" branch hasn't been updated since October:
![image](https://user-images.githubusercontent.com/26899845/118653736-21a9bd80-b7e8-11eb-9c53-dfb63c11ada1.png)

**Comment by AntonBallmaier** [comment]
> CC @XorUnison, you mentioned at some point that you are looking to rework arrows?

Thanks for the tip - He has given me quite a few more ideas :)

**Comment by XorUnison** [comment]
> CC @XorUnison, you mentioned at some point that you are looking to rework arrows?

Correct, although to my shame I will have to first complete Tiling/Honeycomb which uh... I should've done ages ago.
If someone else works on tips instead that's just fine, and there's some improvements here that I didn't plan on doing. Mainly the condensing of the different arrow classes.

As for what I was intending to do, here's the excerpt from our discord PMs:
> My main idea for the tip itself was to break down the thing as much as possible, into 3 simple properties.
> Position: Ranging from 0 to 1 in float it would determine where on the line the tip is
> Rotation: In radians this would determine how much the tip is rotated away from the direction of the line segment it's placed on, and would be 0 by default
> Tip: Since... the tip is basically just a VMobject, might as well cut out the middlemen and directly make it an accessible VMobject

That aside placing tips along lines/arcs and in far more general moving things along anything line-like such as more complicated functions could be condensed into all into one central package of functionality. That'd require some more digging into those and how the community does that kind of stuff currently however.

**Comment by behackl** [comment]
Great! I like all the suggestions here. `geometry` desperately needs be cleaned up. :-)

**Comment by AntonBallmaier** [comment]
Ok - I'll probably start any time soon and keep you up to date how it's coming along :slightly_smiling_face:

**Comment by JeroSQ** [comment]
Hey, I see you want to improve Arrows and stuff, which I think is great. I was wondering if the stroke width scaling could be reviewed beacause I don't think it's working properly. Also, I think it would be nice if the user could choose to scale the tips, but not the stroke_width, or viceversa. At the moment you can only choose to scale everything, or nothing. I believe you can get around this by using the `max_stroke_width_to_length `or `max_tip_length_to_length `ratios but IMO that isn't very intuitive. I can help with all of this stuff I've mentioned since it doesn't seem very complicated, and you seem to already have a lot on your plate.

Here's what I meant about the stroke width scaling (I tried modifying `max_stroke_width_to_length_ratio` but it didn't change anything when scaling):
```
def construct(self):
        a = Arrow()

        self.add(a)
        self.wait()
        self.play(a.animate.scale(3, scale_tips=True))
        self.wait()
```


https://user-images.githubusercontent.com/68753500/120089500-544d9300-c0d1-11eb-8ae9-9cabb5d3188f.mp4

**Comment by hydrobeam** [comment]
Would `Line` also be included in `tipable.py`? It inherits from `TipableVMobject`

**Comment by citrusmunch** [comment]
i'd like to work on this
